### PR TITLE
Yield, YieldFrom, AugAssign propagate taint

### PR DESCRIPTION
**Pull request by bcaller**
Previously `+=` only propagated taint of the last assigment. Now, the original variable is added to `right_hand_side_variables`.

Yield and YieldFrom are treated a bit like augmented assignment, giving the function a final return variable `yld_` a bit like `ret_`. Really it's a generator function, but we can treat it as a normal function. So anything yielded propagates taint to the overall "return" value of the function.

### CODE REVIEW

**Comment by KevinHock** [comment]
Thanks so much for making this, it looks great :)

I'll review in-depth tomorrow as I'm still on-a-roll with my open PR.

File: pyt/cfg/expr_visitor.py [hunk 1]
```diff
@@ -114,11 +114,10 @@ def visit_Yield(self, node):
         label = LabelVisitor()
         label.visit(node)
 
-        try:
-            rhs_visitor = RHSVisitor()
-            rhs_visitor.visit(node.value)
-        except AttributeError:
-            rhs_visitor.result = 'EmptyYield'
+        if node.value is None:
```
Comment (KevinHock) [inline pyt/cfg/expr_visitor.py]: You write such high quality code, but the amazing commit messages is what really separates you from the rest.

File: pyt/cfg/expr_visitor.py [hunk 2]
```diff
@@ -113,22 +114,26 @@ def visit_Yield(self, node):
...
+        LHS = 'yld_' + this_function_name
+        return self.append_node(YieldNode(
+            LHS + ' += ' + label.result,
             LHS,
             node,
-            rhs_visitor.result,
-            path=self.filenames[-1])
+            rhs_visitor_result + [LHS],
+            path=self.filenames[-1],
+            line_number=node.lineno)
```
Comment (KevinHock) [inline pyt/cfg/expr_visitor.py]: So for this I don't think you need to pass `line_number` due to
https://github.com/python-security/pyt/blob/master/pyt/core/node_types.py#L42-L43

**Comment by bcaller** [comment]
done

**Pull request merged**